src/main/resources/theme/icons.css

```diff
@@ -0,0 +1,208 @@
+@import "style.css";
+
+.icon {
+  /* Same color as text painted on the same -fx-background. */
+  -fx-background-color: -fx-text-background-color;
+  /* 16x16px. Non-square icons will have to adjust the padding. */
+  -fx-padding: 0.666667em;
+
+  /* Prevents images from stretching (e.g. in a table cell); but the padding will still give them their space. */
+  -fx-max-height: 0;
+  -fx-max-width: 0;
+}
+
+.window-button .icon {
+  -fx-padding: 0;
+}
+
+.login-logo,
+.icon-faf {
+  -fx-shape: "M16 32c-8.837 0-16-7.163-16-16s7.163-16 16-16 16 7.163 16 16-7.163 16-16 16zM16 2.147c-7.651 0-13.853 6.202-13.853 13.853s6.202 13.853 13.853 13.853 13.853-6.202 13.853-13.853-6.202-13.853-13.853-13.853zM24.026 16.058l3.538 6.127h-2.256l-4.642-8.077h7.122l1.125 1.949h-4.886zM18.256 9.917h7.111l1.24 2.147h-7.117zM20.911 22.185l-1.684-2.917h-6.415l-1.684 2.917h-2.173l7.083-12.268h0.057l7.052 12.268h-2.235zM16.020 13.743l-1.78 3.083h3.56zM5.388 12.064l1.24-2.147h7.243l-1.24 2.147h-7.243zM6.787 22.185h-2.357l3.538-6.127h-4.886l1.125-1.949h7.243z";
+}
```

Comment:
Can u explain me how this works? How do I add anew icon?

Reply:
You can get the SVGPath string from icomoon:
![95929fd1-1255-4f1c-89ab-f00733366a50](https://user-images.githubusercontent.com/1864183/87208751-f6c12400-c30f-11ea-8fc5-68a30f3911c6.gif)

Or from the source code of SVGs. https://raw.githubusercontent.com/FAForever/Art-Assets/master/FAF-Logo.svg

`-fx-shape` will apply the shape to whatever component you have. If it's a button, the button will have the SVG shape. We apply it to Regions like `<Region styleClass="icon,map-size-icon" />`.

The annoying thing about it is that either you set `-fx-scale-shape` to false and the the "original" size as specified in the SVGPath, which is rarely what you want. Or you set `-fx-scale-shape` to true and the shape will adjust to the region's size - without keeping the aspect ratio. So for each icon the dimensions must be specified.

However, the advantages are:
- No need to generate a font
- No need to keep the font and .json file in sync
- No dependency to icomoon
- Themes can easily replace individual icons. Right now, I don't even know how themes can set their own icons. The probably need to provide their own where the icons use the same unicode character, and set this font via CSS only for those nodes whose icon should be changed.
- It's a standard JavaFX mechanism

Reply:
Maybe we should turn this into a wiki page... ?

Reply:
https://github.com/FAForever/downlords-faf-client/wiki/Adding-Icons